subsys/net/ip/net_mgmt.c

```diff
@@ -200,15 +200,24 @@ static int mgmt_event_wait_call(struct net_if *iface,
 				struct net_if **event_iface,
 				int timeout)
 {
+#ifndef __XCC__
```

Comment:
I have a better fix for this. If you put XCC into "-clang -std=c11" mode, a lot of stuff starts working unmodified.
Going to pursue putting the compiler in clang mode.